src/Templates/ModelTemplate.cshtml

```diff
@@ -178,6 +211,11 @@ func (@(Model.Name.FixedValue.ToShortName()) *@(Model.Name)) UnmarshalJSON(body
     var v *json.RawMessage
         @foreach (var p in Model.AllProperties)
         {
+            if (p.ModelType is DictionaryTypeGo && ((DictionaryTypeGo)p.ModelType).SupportsAdditionalProperties)
+            {
+                // TODO: codegen for this scenario is broken
+                continue;
+            }
```

Comment:
is there any swagger that currently has this ? in case that no, does it make sense to throw an exception here to know when that happens ? or do we plan to fix this ASAP ?

Reply:
Yes there are a few that are broken, unless somehow they were fixed by your recent changes to discriminated types.  IIRC @mcardosos was going to look into this?

Reply:
I have not fixed this.
This is happening (at least) in advisor, machine learning and service fabric (dataplane).

Reply:
Opened https://github.com/Azure/autorest/issues/2761 to track this.